okta/oauth.py

```diff
@@ -37,6 +38,13 @@ async def get_access_token(self):
             str, Exception: Tuple of the access token, error that was raised
             (if any)
         """
+        
+        # Check if access token has expired or will expire in the next 5 minutes
+        current_time = int(time.time())
+        if self._access_token and hasattr(self, '_access_token_expiry_time'):
+            if current_time + 300 >= self._access_token_expiry_time:
```

Comment:
I don't want this to be a tough ask, but could this be also present in the config when creating the OktaClient? Have the renewal time also be configurable and default to minutes

Reply:
Happy to take on some feedback.

Just to clarify: You want the renewal time (the number of minutes before the token expires that the token is automatically renewed) to be configurable when creating the Okta Client. Are you happy with default of 5 minutes?

So configuring the client might look something like this:

```python
  configuration = {
      'orgUrl': f'https://{current_app.config["OKTA_BASE_URL"]}',
      'authorizationMode': 'PrivateKey',
      'clientId': current_app.config['OKTA_API_CLIENT_ID'],
      'scopes': current_app.config['OKTA_API_SCOPES'],
      'privateKey': current_app.config['OKTA_API_PRIVATE_KEY'],
      'kid': current_app.config['OKTA_API_PRIVATE_KEY_ID'],
      'oauthTokenRenewalOffset': 10 # Changing the default of 5 minutes to 10 minutes before expiration
  }
  current_app.okta_client = OktaClient(configuration)
```

And the token renewal code would look something like:
```python
current_time = int(time.time())
if self._access_token and hasattr(self, '_access_token_expiry_time'):
    if current_time + self._config['oauthTokenRenewalOffset'] >= self._access_token_expiry_time:
        self.clear_access_token()
```

Reply:
LGTM. The 5 minute default is okay IMO

Reply:
Thanks @haggrip, I've made these changes, and I've integrated the changes more completely into the application's convention/structure - initialising the configuration variable with the Client, setting the default value with the ConfigSetter and applying the default value with _apply_default_values etc.

Hopefully this is what you had in mind. In my manual testing, it works well.